dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -2270,22 +2271,40 @@ public Response handleGitHubInstallation(@ApiParam(hidden = true) @Parameter(hid
         });
 
         if (added) {
-            // make some educated guesses whether we should try to retrospectively release some old versions
+            // for each added repository, try to retrospectively release some old versions.
+            // if we inspected all of the branches and didn't find any that were releasable,
+            // attempt to infer/deliver a .dockstore.yml on the "most important" branch.
             // note that for large organizations, this loop could be quite large if many repositories are added at the same time
             for (String repository: repositories) {
-                final Set<String> strings = identifyGitReferencesToRelease(repository, installationId);
-                for (String gitReference: strings) {
-                    if (LOG.isInfoEnabled()) {
-                        LOG.info(String.format("Retrospectively processing branch/tag %s in %s(%s)", Utilities.cleanForLogging(gitReference), Utilities.cleanForLogging(repository),
-                            Utilities.cleanForLogging(username)));
+                final int maximumBranchCount = 5;
+                final List<String> importantBranches = identifyImportantBranches(repository, installationId);
+                final List<String> releasableReferences = identifyGitReferencesToRelease(repository, installationId, subList(importantBranches, maximumBranchCount));
+                if (releasableReferences.isEmpty()) {
+                    boolean inspectedAllBranches = importantBranches.size() <= maximumBranchCount;
+                    if (inspectedAllBranches) {
+                        if (!importantBranches.isEmpty()) {
+                            String mostImportantBranch = importantBranches.get(0);
```

Comment:
Do we want to allow the user to select a branch to infer if the important branch that we select is not the one they want, potentially leveraging this deprecated endpoint? https://github.com/dockstore/dockstore/blob/9615f4861b862ff18e0b949bc14e78140aed5e76/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java#L2163 (side note, if this endpoint is not going to be used, should we delete it? or update it to point at the newest inference code?)

For example, the user could select a branch to infer on the GitHub App landing page if the one that we generated and display is not the branch they want.

This is just for discussion purposes, and I am okay with the approach here as is